src/log/entry.cr

```diff
@@ -0,0 +1,23 @@
+# A logging severity level.
+enum Log::Severity
+  Debug
+  Verbose
+  Info
+  Warning
+  Error
+  Fatal
+  # Used only for severity level filter
+  None
+end
+
+struct Log::Entry
+  getter source : String
+  getter severity : Severity
+  getter message : String
+  getter timestamp = Time.local
+  getter context = Log.context
+  getter exception : Exception?
```

Comment:
Any chance of including file/line information here? (from #8810) ☺️

Reply:
You can add it as context information if needed.

Explicitly or with a method that will do this.

```crystal
Log.context.using do
  Log.context.set file: __FILE__, line: __LINE__
  Log.info { ... }
end # remove file/line context keys
```

I'm not sure every entry should have information. But I am open to discuss the idea.

Reply:
But why not add file and line information to every Entry?

It's as simple as letting `info`, `debug` and other methods accept `file = __FILE__, line = __LINE__`.

On the other hand, you can usually find where a log line happens by searching that text in code, so I'm not sure how useful it is to include it in every log line.

Reply:
> On the other hand, you can usually find where a log line happens by searching that text in code, so I'm not sure how useful it is to include it in every log line.

@asterite Yeah that's my main reason for not wanting it in every log line. For example, if a shard outputs an error, I don't want/need to know which line it is from, or if it is just general info I probably don't need it either. 

Another option might be to always include file/line like you suggested, but make whether that info is actual set in the context an option:

```crystal
Log.log_file_and_line = true
```

And then the block version if you just want it temporarily as suggested above. That might be pretty cool and very useful if you really do need to figure out where a message is coming from and can't find it

Reply:
The default formatter doesn't have to output this information, but being able access this information in a custom formatter is super useful - especially if the log message is somewhat generic.

The alternative is having to maintain my own wrapper around the standard logger, which is less than ideal.

Reply:
@willhbr That is a very nice idea! May as well always have it available and let the formatter decide. The default formatters could have a configurable option for outputting it (or not, someone can add it later)